src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/FlagAuthoritySystem.java

```diff
@@ -113,4 +120,38 @@ public void onGiveItemToCharacterHoldItem(GiveItemEvent event, EntityRef item, I
                 delayManager.cancelDelayedAction(item, LASUtils.DROPPED_FLAG);
         }
     }
+
+    @ReceiveEvent
+    public void dropFlagRequest(FlagDropRequestEvent event, EntityRef targetPlayer) {
+        int inventorySize = inventoryManager.getNumSlots(targetPlayer);
+        for (int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
+            EntityRef inventorySlot = inventoryManager.getItemInSlot(targetPlayer, slotNumber);
+            if (inventorySlot.hasComponent(BlockItemComponent.class)) {
+                if (inventorySlot.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(event.flagTeam)) {
+                    flagSlot = inventorySlot;
+                }
```

Comment:
you could consider working with optionals here to avoid first checking "has" and then doing "get". with an optional you could do "get" and "ifPresent" continue to get the block family, etc.